### PR TITLE
[GEOT-6524] The GeoPackage Plugin does not support long integers

### PR DESCRIPTION
**Pull request by aaime**
GeoPackage INT are actually longs in range.
See also https://osgeo-org.atlassian.net/browse/GEOT-6524

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [ ] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer.

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):
- [x] There is an issue in [Jira](https://osgeo-org.atlassian.net/projects/GEOT) describing the bug/task/new feature (a notable exemptions is, changes not visible to end users). The ticket is for the GeoTools project, if the issue was found elsewhere it's a good practice to link to the origin ticket/issue.
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [x] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

File: modules/plugin/geopkg/src/test/java/org/geotools/geopkg/GeoPkgTestSetup.java [hunk 1]
```diff
@@ -67,7 +67,7 @@ protected void setUpData() throws Exception {
         String sql = "CREATE TABLE ft1 (id INTEGER PRIMARY KEY, geometry BLOB)";
         run(sql);
 
-        sql = "ALTER TABLE ft1 add intProperty INTEGER";
+        sql = "ALTER TABLE ft1 add intProperty MEDIUMINT";
```
Comment (nmco) [inline modules/plugin/geopkg/src/test/java/org/geotools/geopkg/GeoPkgTestSetup.java:70]: Not clear to me why this one was changed, I mean I understand that a `` MEDIUMINT`` is an ``int``, bu did the test just keep working whit this change?

**Comment by nmco** [comment]
Build failure looks unrelated:
```
[ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 35.053 s <<< FAILURE! - in org.geotools.ows.wmts.online.WMTSServerTest
[ERROR] testWMTSCoverageReader(org.geotools.ows.wmts.online.WMTSServerTest)  Time elapsed: 34.029 s  <<< FAILURE!
java.lang.AssertionError: 
Images are visibly different, found 128 different pixels, against a threshold of 50
You can add -Dorg.geotools.image.test.interactive=true to show a dialog comparing them (requires GUI support)
	at org.geotools.ows.wmts.online.WMTSServerTest.testWMTSCoverageReader(WMTSServerTest.java:146)
```

I restarted them.

**Comment by nmco** [comment]
Small question, apart from that looks good!

File: modules/plugin/geopkg/src/test/java/org/geotools/geopkg/GeoPkgTestSetup.java (hunk 1 above)
Comment (aaime) [reply modules/plugin/geopkg/src/test/java/org/geotools/geopkg/GeoPkgTestSetup.java:70]: MEDIUMINT is a 32 bits integer, INTEGER is a 64bit integer. The map to different types in Java now (the previous code was not aware of the various types of sqlite integers, from 8 to 64 bits)

**Pull request merged**